.github/workflows/update-snippets.yml

```
@@ -0,0 +1,56 @@
+name: Update Snippets
+# Update cylc-snippets.json using bin/cylc-snippets.
+
+on:
+  # Ideally people making changes should re-run this task, but probably
+  # won't remember - I think once a month is probably often enough.
+  schedule:
+    - cron: "51 03 15 * *" # 03:51 on the 15th of the month
+  workflow_dispatch:
+
+jobs:
+  update-snippets:
+    runs-on: ubuntu-latest
+    outputs:
+      create_pr: ${{ steps.set_output.outputs.create_pr }}
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install Cylc
+      run: pip install cylc-flow
+
+    - name: Update Snippets
+      run: ./bin/get_keywords > cylc-snippets.json
+
+    - name: Configure git
+      uses: cylc/release-actions/configure-git@v1
+
+    - name: git diff and set output
+      id: set_output
+      run: |
+        if [[ -n $(git status cylc-snippets.json) ]]; then
+          echo "create_pr=true" >> $GITHUB_OUTPUT
+        else
+          echo "create_pr=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Checkout PR branch
+      if: steps.set_output.outputs.create_pr == 'true'
+      run: |
+        git checkout -b "update_snippets"
+
+    - name: Commit & push
+      if: steps.set_output.outputs.create_pr == 'true'
+      run : |
+        git commit -a -m "Update Cylc Config Snippets" -m "Workflow: ${{ github.workflow }}, run: ${{ github.run_number }}"
+        git push origin update_snippets -f
```

I'm not sure it should force push? 

Btw, no need to specify the remote and branch because you have used `cylc/release-actions/configure-git`
```suggestion
        git push
```